openassessment/workflow/models.py

```diff
@@ -239,7 +240,8 @@ def get_score(self, assessment_requirements, step_for_name):
                     else:
                         requirements = assessment_requirements.get(assessment_step_name, {})
                     score = get_score_func(self.submission_uuid, requirements)
-                    break
+                    if score:
+                        break
```

Comment:
This may not work. If you have a question with Peer + Self, and you complete your Peer section, then Self, then you should be in a state where you are waiting for assessments from your Peer. With this change, I think you may end up just giving them their "Self" score. 

I think you may need to change all the APIs to return a score, as well as whether the score should not be omitted.   This may be worth updating the get_score function to return another attribute in the dictionary, something like optional: True